tests/MediaTest.php

```diff
@@ -0,0 +1,334 @@
+    {
+        $this->mockGenerateFilename();
+
+        $response = $this->actingAs(self::$admin)->json('post', '/media', ['file' => self::$file]);
+
+        $response->assertCreated();
+
+        $this->assertDatabaseHas('media', [
+            'id' => 6,
+            'link' => '/storage/file.png',
+        ]);
+
+        Storage::disk('local')->assertExists($this->getFilePathFromUrl('file.png'));
+
+        $this->clearUploadedFilesFolder();
+    }
+
+    public function testCreateNoAuth(): void
+    {
+        $response = $this->json('post', '/media', ['file' => self::$file]);
+
+        $response->assertUnauthorized();
+    }
+
+    public function testBulkCreate(): void
+    {
+        $this->mockGenerateFilename('file1.png', 'file2.png');
+
+        $response = $this->actingAs(self::$admin)->json('post', '/media/bulk', [
+            'media' => [
+                [
+                    'file' => self::$file,
+                    'meta' => ['test1'],
+                ],
+                [
+                    'file' => self::$file,
+                    'meta' => ['test2'],
+                ],
+            ],
+        ]);
+
+        $response->assertOk();
+
+        $this->assertDatabaseHas('media', [
+            'id' => 6,
+            'name' => 'file1.png',
+            'owner_id' => self::$admin->id,
+            'meta' => "[\"test1\"]",
+            'is_public' => false,
+        ]);
+
+        $this->assertDatabaseHas('media', [
+            'id' => 7,
+            'name' => 'file2.png',
+            'owner_id' => self::$admin->id,
+            'meta' => "[\"test2\"]",
+            'is_public' => false,
+        ]);
+    }
+
+    public function testDelete(): void
+    {
+        $response = $this->actingAs(self::$admin)->json('delete', '/media/1');
+
+        $response->assertNoContent();
+
+        $this->assertSoftDeleted('media', [
+            'id' => 1,
+        ]);
+    }
+
+    public function testDeleteNotExists(): void
+    {
+        $response = $this->actingAs(self::$admin)->json('delete', '/media/0');
+
+        $response->assertNotFound();
+    }
+
+    public function testDeleteNoPermission(): void
+    {
+        $modelTestState = new ModelTestState(Media::class);
+
+        $response = $this->actingAs(self::$user)->json('delete', '/media/1');
+
+        $response->assertForbidden();
+
+        $modelTestState->assertNotChanged();
+    }
+
+    public function testDeleteNoAuth(): void
+    {
+        $modelTestState = new ModelTestState(Media::class);
+
+        $response = $this->json('delete', '/media/1');
+
+        $response->assertUnauthorized();
+
+        $modelTestState->assertNotChanged();
+    }
+
+    public function getSearchFilters(): array
+    {
+        return [
+            [
+                'filter' => ['all' => true],
+                'result' => 'get_by_all.json',
+            ],
+        ];
+    }
+
+    public function getAdminSearchFilters(): array
```

Comment:
admin has the same filters logic as owner

Reply:
Got rid of the admin user in tests, since the roles were previously removed.